src/plugins/intel_cpu/src/nodes/executors/mlas/mlas_transpose.cpp

```diff
@@ -0,0 +1,299 @@
+    auto writes_per_writer_per_loop = int64_t(writes_per_loop / num_writers);
+    // TODO: check integer overflow
+    const size_t bytes_per_write = static_cast<size_t>(block_size) * element_size;
+
+    switch (bytes_per_write) {
+        case (sizeof(uint8_t)): {
+            SimpleTransposeSingleAxisOutwards(input_data, output_data, num_loops, num_writers, writes_per_loop,
+                                                                                writes_per_writer_per_loop);
+            break;
+        }
+        case (sizeof(uint16_t)): {
+            SimpleTransposeSingleAxisOutwards(reinterpret_cast<const uint16_t*>(input_data),
+                                                                                reinterpret_cast<uint16_t*>(output_data), num_loops, num_writers,
+                                                                                writes_per_loop, writes_per_writer_per_loop);
+            break;
+        }
+        case (sizeof(uint32_t)): {
+            SimpleTransposeSingleAxisOutwards(reinterpret_cast<const uint32_t*>(input_data),
+                                                                                reinterpret_cast<uint32_t*>(output_data), num_loops, num_writers,
+                                                                                writes_per_loop, writes_per_writer_per_loop);
+            break;
+        }
+        case (sizeof(uint64_t)): {
+            SimpleTransposeSingleAxisOutwards(reinterpret_cast<const uint64_t*>(input_data),
+                                                                                reinterpret_cast<uint64_t*>(output_data), num_loops, num_writers,
+                                                                                writes_per_loop, writes_per_writer_per_loop);
+            break;
+        }
+        default: {
+            // we need to use memcpy for each block
+            for (int64_t l = 0; l < num_loops; ++l) {
+                uint8_t* output_for_first_writer = output_data;
+
+                for (auto wwpl = 0; wwpl < writes_per_writer_per_loop; ++wwpl) {
+                    uint8_t* output_for_current_writer = output_for_first_writer;
+
+                    for (uint64_t w = 0; w < num_writers; ++w) {
+                        memcpy(output_for_current_writer, input_data, bytes_per_write);
+                        // skip to output position for next writer
+                        output_for_current_writer += (writes_per_writer_per_loop * bytes_per_write);
+                        input_data += bytes_per_write;
+                    }
+                    output_for_first_writer += bytes_per_write;
+                }
+                output_data += writes_per_loop * bytes_per_write;
+            }
+        }
+    }
+}
+
+void MlasTransposeExecutor::TransposeSingleAxisInwards(const MemoryCPtr& input, const MemoryPtr& output, size_t from, size_t to) {
```

Comment:
maybe I missed smt, but I don't see any parallelization in the code. Seems like we will need to evaluate this as follow-up task.